docs/source/nodes_and_pipelines/nodes.md

```diff
@@ -184,114 +184,51 @@ You can also call a node as a regular Python function: `adder_node(dict(a=2, b=3
-
-        super().__init__(
-            filepath=PurePosixPath(path),
-            version=version,
-            exists_function=self._fs.exists,
-            glob_function=self._fs.glob,
-        )
-
-        # Handle default load and save arguments
-        self._load_args = deepcopy(self.DEFAULT_LOAD_ARGS)
-        if load_args is not None:
-            self._load_args.update(load_args)
-        self._save_args = deepcopy(self.DEFAULT_SAVE_ARGS)
-        if save_args is not None:
-            self._save_args.update(save_args)
-
-    def _describe(self) -> Dict[str, Any]:
-        return {
-            "filepath": self._filepath,
-            "protocol": self._load_args,
-            "save_args": self._save_args,
-            "version": self._version,
-        }
-
-    def _load(self) -> pd.DataFrame:
-        load_path = str(self._get_load_path())
-        return pd.read_csv(load_path, **self._load_args)
-
-    def _save(self, data: pd.DataFrame) -> None:
-        save_path = get_filepath_str(self._get_save_path(), self._protocol)
+Create the project with this command:
+```bash
+kedro new -s pandas-iris
+```
 
-        buf = BytesIO()
-        data.to_csv(path_or_buf=buf, **self._save_args)
+### Loading data with Generators
+To use generator functions in Kedro nodes, you need to update the `catalog.yml` file to include the `chunksize` argument for the relevant dataset that will be processed using the generator.
 
-        with self._fs.open(save_path, mode="a+") as fs_file:
-            fs_file.write(buf.getvalue())
+You need to add a new dataset in your `catalog.yml` as follows:
+```diff
++ X_test:
++  type: pandas.CSVDataSet
++  filepath: data/05_model_input/X_test.csv
++  load_args:
++    chunksize: 10
 ```
-</details>
 
-Modify `example_iris_data` in `catalog.yml` by changing `type` to the custom dataset you created above. Add `chunksize: 100` to `load_args` which will return an iterable object. The `chunksize` parameter refers to the number of rows in each chunk.
+With `pandas` built-in support, you can use the `chunksize` argument to read data using generator.
 
-```yaml
-example_iris_data:
-  type: YOUR_PROJECT_NAME.extras.datasets.chunkwise_dataset.ChunkWiseCSVDataSet
-  filepath: data/01_raw/iris.csv
-  load_args:
-    chunksize: 100
-```
+### Saving data with Generators
+To use generators to save data lazily, you need do three things:
+- Update the `make_prediction` function definition to use `return` instead of `yield`.
+- Create a [custom dataset](../extend_kedro/custom_datasets.md) called `ChunkWiseCSVDataset`
+- Update `catalog.yml` to use a newly created `ChunkWiseCSVDataset`.
 
-Next, in `nodes.py` we repurpose the existing `split_data` function to process chunk-wise data:
+Copy the following code to `nodes.py`. The main change is to use a new model `DecisionTreeClassifier` to make prediction by chunks in `make_predictions`.
 
+<details><summary>Click to open
```

Comment:
The formatting doesn't look quite right when you open the file: https://github.com/kedro-org/kedro/blob/899e0c59d51d1f8de746b9ae662c5c4cc3556164/docs/source/nodes_and_pipelines/nodes.md